### ISSUE TITLE
`IgnoredReturnValue` false positives for functions that return Unit

### ISSUE THREAD

**Issue by UnguidedFreedom** [detekt/detekt#9183]
## Expected Behavior
A function call returning `Unit` should not be flagged by `IgnoredReturnValue`

## Observed Behavior
The rule gets triggered

## Steps to Reproduce
```kotlin
fun demo(log: (String) -> Unit): String {
    val text = "Hello"
    log(text)
    return text
}
```
leads to
```
The call log is returning a value that is ignored. [IgnoredReturnValue]
```

## Context
Forcing to put a bunch of `@Suppress` when using functions returning `Unit`

## Your Environment

* Version of detekt used: 2.0.0-alpha.2
* Version of Gradle used (if applicable): 9.4.1
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: macOS 26.3
* Link to your project (if it's a public repository): N/A

**Comment by atulgpt** [comment]
We might retire this rule in favour of Kotlin handling of return values. See the issue at https://github.com/detekt/detekt/issues/8932

**Comment by BraisGabin** [comment]
This should be fixed anyway. I don't see that rule disappearing soon.
